libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java

```diff
@@ -223,9 +220,6 @@ public void onPause() {
 
   public void onStop() {
     mapView.onStop();
-    if (navigationMap != null) {
-      navigationMap.onStop();
-    }
```

Comment:
Does onStop need lifecycleRegistry.markState(Lifecycle.State.STOPPED);  as I see in onStart?

Reply:
This may be outside of the scope of your PR so I don't think merging depends on this necessarily but it caught my attention so if we could at least create a ticket to investigate whether lifecycleRegistry.markState(Lifecycle.State.STOPPED) is needed that would be great so we don't lose track of it.

Besides that, I reviewed the changes again. I think this is the right way to go and I think the changes should be merged in.

Reply:
@cafesilencio definitely it's not a right way how NavigationView should work 👍 
I've created separate issue https://github.com/mapbox/mapbox-navigation-android/issues/3024 for missing calls. Would be great investigate and fix it in separate PR because here might be more than one missing call(have a look on `fun onPause()`, missing call as well)
If everything is ok I need ✅